Modules/_asynciomodule.c

```diff
@@ -594,11 +596,33 @@ future_set_exception(FutureObj *fut, PyObject *exc)
     Py_RETURN_NONE;
 }
 
+static PyObject *
+create_cancelled_error(PyObject *msg)
+{
+    PyObject *exc;
+    if (msg == NULL || msg == Py_None) {
+        msg = PyUnicode_FromString("");
+        exc = PyObject_CallOneArg(asyncio_CancelledError, msg);
```

Comment:
Why not `_PyObject_CallNoArg`? That would be simpler as you wouldn't need to do the `PyUnicode_FromString("");` dance.

Reply:
That's how I had it originally, but I changed it to the above after making this change:

```python
raise exceptions.CancelledError(
  '' if self._cancel_message is None else self._cancel_message)
```

I have tests to check that the created exception object is the same C and Python implementations (`CancelledError()` vs `CancelledError('')`). I think it's good to have the behavior match when possible, even if subtle.

Before, the exception helper I had in the Python code was doing the logic you suggest here.